include/eve/module/core/regular/impl/simd/x86/convert.hpp

```diff
@@ -10,8 +10,754 @@
+    {
+      auto w = _mm_packs_epi32(v, v);
+      if constexpr(sizeof(U) == 1) return _mm_packs_epi16(w, w);
+      else return w;
+    }
+    else  if constexpr( c_in == category::float32x4  )
+    {
+            if constexpr( sizeof(U) < 4 )
+      {
+        return convert( convert(v,as<logical<as_integer_t<T,signed>>>{}), tgt);
+      }
+      else  if constexpr( match(c_out, category::int64x2, category::uint64x2) )
+      {
+        return convert( convert(v,as<logical<std::int32_t>>{}), tgt);
+      }
+      else  if constexpr( sizeof(U) == 8 && N::value == 4 )
+      {
+        return eve::combine(convert( v.slice(lower_), tgt), convert( v.slice(upper_), tgt));
+      }
+      else return convert_impl(EVE_RETARGET(cpu_),v,tgt);
+    }
+    else  if constexpr( N::value == 8 && sizeof(T) == 4  )
+    {
+      return eve::combine(convert( v.slice(lower_), tgt), convert( v.slice(upper_), tgt));
+    }
+    //==============================================================================================
+    // 64 cases
+    //==============================================================================================
+    else  if constexpr( sizeof(T) == 8 && sizeof(U) < 4 )
+    {
+      return convert( convert(v,as<logical<downgrade_t<as_integer_t<T,signed>>>>{}), tgt);
+    }
+    else  if constexpr(   c_in == category::float64x2
+                      && match(c_out, category::int64x2, category::uint64x2)
+                      )
+    {
+      return convert( convert(v,as<logical<float>>{}), tgt);
+    }
+    else  if constexpr( c_in == category::int64x2 )
+    {
+      using i_t = as<logical<std::int32_t>>;
+
+      if constexpr(match(c_out, category::float64x2, category::float32x4))
+        return convert( convert(v,i_t{}), tgt);
+      else
+        return convert_impl(EVE_RETARGET(cpu_),v,tgt);
+    }
+    else if constexpr( c_in == category::int64x4 && c_out == category::float32x4 )
+    {
+      return convert( convert(v,as<logical<std::int32_t>>{}), tgt);
+    }
+    //==============================================================================================
+    // Generic case
+    //==============================================================================================
+    else
+    {
+      return convert_impl(EVE_RETARGET(cpu_),v,tgt);
+    }
+  }
+
+  //================================================================================================
+  // convert: float64 -> U
+  //================================================================================================
+  template<typename N, real_scalar_value U>
+  EVE_FORCEINLINE wide<U, N> convert_impl ( EVE_SUPPORTS(sse2_)
+                                          , wide<double,N> v, as<U> const &tgt
+                                          ) noexcept
+  {
+    constexpr auto c_i = categorize<wide<double, N>>();
+    constexpr auto c_o = categorize<wide<U, N>>();
+
+          if constexpr( c_i == category::float64x2 )
```

Comment:
You don't need the input category split - the output category will be sufficient.

Reply:
I think I do cause I need to discriminate between float64x2 -> float32x4 (only 2 lanes used) and float64x4 -> float32x4

Reply:
Reworked the if constexpr layout to be more flat